sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -142,12 +140,14 @@ object ScanOperation extends OperationHelper with PredicateHelper {
       case Filter(condition, child) =>
         collectProjectsAndFilters(child) match {
           case Some((fields, filters, other, aliases)) =>
-            // Follow CombineFilters and only keep going if the collected Filters
-            // are all deterministic and this filter doesn't have common non-deterministic
-            // expressions with lower Project.
-            if (filters.forall(_.deterministic) &&
-              !hasCommonNonDeterministic(Seq(condition), aliases)) {
-              val substitutedCondition = substitute(aliases)(condition)
+            // Follow CombineFilters and only keep going if 1) the collected Filters
+            // and this filter are all deterministic or 2) if this filter is the first
+            // collected filter which is non-deterministic and doesn't have common
```

Comment:
`... which is non-deterministic ...` this is not corrected. The first collected filter can be non-deterministic

Reply:
You mean it can also be deterministic? Is it a typo?

Reply:
ah sorry I misread it. The only thing that matters is: it should be the first collected filter. It doesn't have to be deterministic or non-deterministic.

Reply:
I see.

Reply:
can we update the comment?